src/pages/HomePage/HomePage.tsx

```diff
@@ -0,0 +1,98 @@
+import Header from '../../components/Header/Header.tsx';
+import Main from '../../components/Main/Main.tsx';
+import { fetchAllPokemon, fetchPokemonPage } from '../../utils/api.ts';
+import type { IPokemon } from '../../utils/types.ts';
+import { useEffect, useState } from 'react';
+import { Outlet, useSearchParams } from 'react-router';
+import { useLocalStorage } from '../../Hooks/useLocalStorage.ts';
+import Pagination from '../../components/Pagination/Pagination.tsx';
+
+export default function HomePage() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [storedSearch, setStoredSearch] = useLocalStorage('searchTerm', '');
+
+  const [pokemonList, setPokemonList] = useState<IPokemon[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const limit = 8;
+  const page = parseInt(searchParams.get('page') || '1', 10);
+  const offset = (page - 1) * limit;
+
+  const search = searchParams.get('search') ?? storedSearch;
+
+  const totalPages = search ? 1 : Math.ceil(1302 / limit);
+
+  const handleSearch = (term: string) => {
+    setStoredSearch(term);
+    searchParams.set('search', term);
```

Comment:
Instead of 
`searchParams.set('search', term);
    searchParams.set('page', '1');
    setSearchParams(searchParams);`

try:
`setSearchParams((current) => {
      const params = new URLSearchParams(current);
      if (term) {
        params.set('search', term);
      } else {
        params.delete('search');
      }
      params.set('page', '1');
      return params;
    });`

It will not empty `search` in the params list